### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by daisieh**
While working on https://candig.atlassian.net/browse/DIG-1105, I did a lot of refactoring of code for clarity. For ease of review, I split up the changes: this PR just does the refactor, with minimal actual code changes. The test suite has no new tests, but should still run successfully as before:

```
cd lib/htsget-server/htsget_app/
source ../../../env.sh
export TESTENV_URL=http://docker.localhost:3333
pytest
```

### CODE REVIEW

File: htsget_server/authz.py [hunk 1]
```diff
@@ -21,16 +23,11 @@ def is_authed(id_, request):
         return 200
     if "Authorization" in request.headers:
         authed_datasets = get_authorized_datasets(request)
-        obj, code2 = drs_operations.get_object(id_)
-        if code2 == 200:
+        obj= database.get_drs_object(id_)
```
Comment (OrdiNeu) [inline htsget_server/authz.py]:
```suggestion
        obj = database.get_drs_object(id_)
```

**Comment by daisieh** [comment]
poke for travis

File: htsget_server/drs_operations.py [hunk 2]
```diff
@@ -38,7 +37,11 @@ def get_object(object_id, expand=False):
     access_url_parse = re.match(r"(.+?)/access_url/(.+)", escape(object_id))
     if access_url_parse is not None:
         return get_access_url(access_url_parse.group(1), access_url_parse.group(2))
-    new_object = database.get_drs_object(escape(object_id), expand)
+    if object_id is not None:
+        new_object = database.get_drs_object(escape(object_id), expand)
+        auth_code = authz.is_authed(escape(object_id), request)
+        if auth_code != 200:
+            return {"message": f"Not authorized to access object {object_id}"}, auth_code
     if new_object is None:
```
Comment (OrdiNeu) [inline htsget_server/drs_operations.py:46]: I think you might get a `'new_object' referenced before assignment here` error if `object_id` is None above.

**Review by OrdiNeu — CHANGES_REQUESTED** [review]
Minor changes -- I'm uncertain how to trigger the error I mentioned above from outside but I'm pretty confident that it's a regression. Once fixed it should be good to go through.

File: htsget_server/drs_operations.py (hunk 2 above)
Comment (daisieh) [reply htsget_server/drs_operations.py:46]: fixed

**Review by OrdiNeu — APPROVED** [review]

**Pull request merged**